vault/core.go

```diff
@@ -2687,6 +2697,10 @@ func (c *Core) adjustForSealMigration(unwrapSeal Seal) error {
 			// We have the same barrier type and the unwrap seal is nil so we're not
 			// migrating from same to same, IOW we assume it's not a migration.
 			return nil
+		case c.seal.BarrierType() == wrapping.WrapperTypeShamir && isUnsealRecoverySeal(c.seal):
```

Comment:
When does it make sense to migrate to a recovery seal?

Reply:
It doesn't, but adjustForSealMigration is called regardless of whether a migration is occurring or not, and without allowing this case, it rejects the seal in recovery mode and errors out.

Reply:
Mind adding a comment?

Reply:
sure